pkg/personname/info_test.go

```diff
@@ -0,0 +1,403 @@
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+			},
+			NoNullSeparators: false,
+			IsEmpty:          false,
+		},
+		// No Ideographic, no seps
+		{
+			Raw: "aFamily^aGiven^aMiddle^aPrefix^aSuffix",
+			Expected: Info{
+				Alphabetic: GroupInfo{
+					FamilyName:       "aFamily",
+					GivenName:        "aGiven",
+					MiddleName:       "aMiddle",
+					NamePrefix:       "aPrefix",
+					NameSuffix:       "aSuffix",
+					NoNullSeparators: true,
+				},
+				Ideographic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+				Phonetic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+			},
+			NoNullSeparators: true,
+			IsEmpty:          false,
+		},
+		// Empty with seps
+		{
+			Raw: "=" +
+				"=" +
+				"",
+			Expected: Info{
+				Alphabetic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+				Ideographic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+				Phonetic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+			},
+			NoNullSeparators: false,
+			IsEmpty:          true,
+		},
+		// Empty no seps
+		{
+			Raw: "",
+			Expected: Info{
+				Alphabetic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+				Ideographic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+				Phonetic: GroupInfo{
+					FamilyName:       "",
+					GivenName:        "",
+					MiddleName:       "",
+					NamePrefix:       "",
+					NameSuffix:       "",
+					NoNullSeparators: true,
+				},
+			},
+			NoNullSeparators: true,
+			IsEmpty:          true,
+		},
+	}
+
+	for _, tc := range testCases {
+
+		// Test creating a new Info object.
+		t.Run(tc.Raw+"_new", func(t *testing.T) {
+			newInfo := Info{
+				Alphabetic:       tc.Expected.Alphabetic,
+				Ideographic:      tc.Expected.Ideographic,
+				Phonetic:         tc.Expected.Phonetic,
+				NoNullSeparators: tc.NoNullSeparators,
+			}
+
+			if tc.Raw != newInfo.String() {
+				t.Errorf(
+					"formatted string: expected '%v', got '%v'",
+					tc.Raw,
+					newInfo.String(),
+				)
+			}
+		})
```

Comment:
is this test case needed? It seems like it's testing that String() is equivalent to the Raw? If so, maybe we should rename the test case comment and name to indicate that.